### ISSUE TITLE
Details of the evaluation method

### ISSUE THREAD

**Issue by niazangels** [jingxil/Pointer-Networks#1]
Hello jingxil,

Thank you for sharing this!
This is one of the best, most understandable implementations of Pointer Networks I have come across. :smiley: 

I was able to train the model and achieve some good training results:
![ch-5](https://user-images.githubusercontent.com/2761491/28438980-aef39f9a-6dbd-11e7-8de8-cb264c5e3a8e.png)

Could you please help me with the following questions? 

1. Is the purpose of the **Forward Only** mode to only switch to evaluation mode, or is there something more?
2. During prediction how are encoder and decoder weights decided? During training we maintain an array of weights for the encoder (shape: [batch_size,max_input_sequence_len,2]) and  decoder (shape: [batch_size,max_output_sequence_len])

    What happens when we see a new input during prediction? Do we set the encoder weights and decoder weights to '1' with zero padding?

Thanks again for the amazing work!

**Comment by jingxil** [comment]
Hi, niazangels

Thank you. I am glad to see you successfully ran the model.  For those questions:
1. Yes, the **Forward Only** flag is only used to switch between training and evaluation. (FYI. I have implemented and uploaded the evaluation part )
2. During prediction, the model is only fed with **inputs** and **enc_input_weights**. And then the model continuously decodes until \<END\> token is predicted. The **enc_input_weights** can be calculated and **dec_input_weights** is not used.
3. I prefer to think in this way: the input sequence is always [0,1,2,...,N] and output at every time step is picked from the input. So, there is no 'new' input. We do set weights to '1' with zero padding.

**Comment by niazangels** [comment]
Thanks for the fantastic explanation, and updates!

I spent all day today reading through the code and was trying to implement a prediction method myself. Got a little confused about the size of the `encoder_input_ids`, and was about to raise a question here and was delighted to see an update with a correction and an eval method.

I have a couple more questions for you: 
1. You mention that the code creates a Pointer Network that is "almost the same" as the one mentioned in the paper. Is the change restricted to the probability function? Or is there something more?

2. Does this customization restrict us from reusing the `PointerNet` class for solving the other problems mentioned in the paper (TSP and Delaunay triangulation) ? The authors have mentioned they used the same model hyperparameters to produce results.

**Comment by jingxil** [comment]
1. In the paper, only one layer of LSTM is used. In this implementation, stacking LSTM layers is allowed. That is why I said "almost the same".
2. No, you can definitely reuse it to solve other problems.

**Comment by saniahamid** [comment]
How will the of attention mechanism change, if you want to the model to solve Delaunay Triangulation?
